docs/hybrid_applications.md

```diff
@@ -35,39 +35,121 @@ If you are sure that you don't call the native SDK before the React Native SDK t
 
-#### Potential issues
+On iOS, use a `UIKitRUMViewsPredicate` to filter out native views created by your navigation libraries:
 
-**You won't have auto instrumentation for JS errors, resources or interactions.**
+```swift
+class RNHybridPredicate: UIKitRUMViewsPredicate {
+    var defaultPredicate = DefaultUIKitRUMViewsPredicate()
 
-You can mitigate this by starting the auto-instrumentation by copying [what the `enableFeatures` function does][4] at the start of your React Native application.
+    func rumView(for viewController: UIViewController) -> RUMView? {
+        let canonicalClassName = NSStringFromClass(type(of: viewController))
+        // Dropping RN Views
+        if (canonicalClassName.starts(with: "RN")) {
+            return nil
+        }
 
-You must use imports to the full path to some of the files (for example, `import {DdRumErrorTracking} from '@datadog/mobile-react-native/src/rum/instrumentation/DdRumErrorTracking';`), and the path to these files might change across SDK versions:
+        return defaultPredicate.rumView(for: viewController)
+    }
+}
 
-```javascript
-import { DdRumErrorTracking } from '@datadog/mobile-react-native/src/rum/instrumentation/DdRumErrorTracking';
-import { DdRumUserInteractionTracking } from '@datadog/mobile-react-native/src/rum/instrumentation/DdRumUserInteractionTracking';
-import { DdRumResourceTracking } from '@datadog/mobile-react-native/src/rum/instrumentation/resourceTracking/DdRumResourceTracking';
-
-DdRumUserInteractionTracking.startTracking();
-DdRumResourceTracking.startTracking({
-    tracingSamplingRate: 80, // Percentage of tracing integrations for network calls between your app and your backend.
-    firstPartyHosts: ['api.example.com'] // Specify the hosts of your backends to enable tracing with these backends
-});
-DdRumErrorTracking.startTracking();
+// Use it when calling trackUIKitRUMViews
+Datadog.Configuration.trackUIKitRUMViews(using: RNHybridPredicate())
 ```
 
-### Initialize the SDK on both sides
+On Android, use a `ComponentPredicate` to filter out native views created by your navigation libraries:
+
+```kotlin
+// Adapt the Fragment type to your View tracking strategy
+class RNComponentPredicate : ComponentPredicate<Fragment> {
+    override fun accept(component: Fragment): Boolean {
+        // Identify and drop react native screen views
+        if (component.javaClass.toString().startsWith("class com.swmansion.rnscreens")) {
+            return false
+        }
+        if (component.javaClass.toString().startsWith("class com.facebook.react")) {
+            return false
+        }
+        return true
+    }
+
+    override fun getViewName(component: Fragment): String? {
+        return null
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+        return true
+    }
+
+    override fun hashCode(): Int {
+        return javaClass.hashCode()
+    }
```

Comment:
I think we don't need these 2. Yes, there are such methods for built-in implementations of `ComponentPredicate` (they were added in https://github.com/DataDog/dd-sdk-android/pull/521), but if I remember correctly they were added for the proper `Configuration.RUM` comparison in tests. Implementations of `ComponentPredicate` in nightly tests are overriding these methods for example.

/cc @xgouchet